src/pkg/util/env/clean.go

```diff
@@ -43,6 +44,21 @@ func SetContainerEnv(g *generate.Generator, env []string, cleanEnv bool, homeDes
 	g.AddProcessEnv("HOME", homeDest)
 	g.AddProcessEnv("PATH", "/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin")
 
+	if prependPath := os.Getenv("SINGULARITYENV_PREPEND_PATH"); prependPath != "" {
+		g.AddProcessEnv("SING_USER_DEFINED_PREPEND_PATH", prependPath)
+		os.Unsetenv("SINGULARITYENV_PREPEND_PATH")
```

Comment:
why are you unseting the ENV VAR after taking it value? is what users expect?

Reply:
@ArangoGutierrez - As I understand it, this is so we don't end up with a `PREPEND_PATH` env var propagated to the container. Users probably don't want that... they just care about `PATH` being assembled as expected.

Reply:
then it can be set here
```go
var alwaysPassKeys = map[string]bool{
	"TERM":        true,
	"http_proxy":  true,
	"https_proxy": true,
	"no_proxy":    true,
	"all_proxy":   true,
	"ftp_proxy":   true,
        "PREPEND_PATH" : false,
}
```

Reply:
That wouldn't work.... `addIfReq` adds everything with a `SINGULARITY_ENV` prefix *before* it considers other things in `alwaysPassKeys`

Reply:
ok then 
```go
	if prependPath := os.Getenv("SINGULARITYENV_PREPEND_PATH"); prependPath != "" {
		g.AddProcessEnv("SING_USER_DEFINED_PREPEND_PATH", prependPath)
		os.Unsetenv("SINGULARITYENV_PREPEND_PATH")
	}

	if appendPath := os.Getenv("SINGULARITYENV_APPEND_PATH"); appendPath != "" {
		g.AddProcessEnv("SING_USER_DEFINED_APPEND_PATH", appendPath)
		os.Unsetenv("SINGULARITYENV_APPEND_PATH")
	}

	if userPath := os.Getenv("SINGULARITYENV_PATH"); userPath != "" {
		g.AddProcessEnv("SING_USER_DEFINED_PATH", userPath)
		os.Unsetenv("SINGULARITYENV_PATH")
	}
```
should be before the `for` loop and not after

Reply:
That's a good catch @ArangoGutierrez

Reply:
yup, good catch @ArangoGutierrez - I'll go get my coffee and reading glasses :-D